support/install.sh

```diff
@@ -0,0 +1,232 @@
+#!/bin/sh
+# Copyright 2024 Northern.tech AS
+#
+#    Licensed under the Apache License, Version 2.0 (the "License");
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+
+ARCH=""
+# TODO: Should we default to the latest from Github?
+VERSION="v1.0.1"
+INSTALL_DIR=${INSTALL_DIR:-"/var/lib/nt-connect"}
+SESSION_TOKEN="${SESSION_TOKEN}"
+OS="linux" # No other options at this stage
+HAS_SYSTEMD="false"
+
+COMMAND="install"
+
+init_defaults() {
+	if test -z "$ARCH"; then
+		local machine=$(uname -m)
+		case "$machine" in
+			x86_64*|amd64*)
+				ARCH="amd64"
+				;;
+			aarch64*|armv8*)
+				ARCH="arm64"
+				;;
+			armv7*)
+				# ???
+				ARCH="armv7"
+				;;
+			armv6*)
+				ARCH="armv6"
+				;;
+			*)
+				printf 'Unsupported machine type "%s"\n' "$machine" 1>&2
+				exit 1
+				;;
+		esac
+	fi
+}
+
+print_usage() {
+	cat 1>&2 <<EOF
+Usage:
+	${0} [--version|-v <version>] [--arch|-a <arch>] [--help|-h]
+
+Arguments:
+	--arch, -a ARCH		Override default system ARCHitecture.
+	--version, -v VERSION	VERSION of nt-connect to install.
+	--help, -h		Display this help text.
+EOF
+}
+
+parse_args() {
+	while test $# -gt 0; do
+		case "$1" in
+			"--version"|"-v")
+				shift
+				VERSION=$1
+				;;
+			"--arch"|"-a")
+				shift
+				ARCH=$1
+				;;
+			"--uninstall"|"-u")
+				COMMAND="uninstall"
+				;;
+			"--help"|"-h")
+				print_usage
+				exit 0
+				;;
+			*)
+				printf 'Invalid argument "%s"\n' "$1" 1>&2
+				print_usage
+				exit 1
+				;;
+		esac
+		shift
+	done
+}
+
+check_dependencies() {
+	if ! jq < /dev/null > /dev/null; then
+		echo "This script needs jq to configure your device" 1>&2
+		exit 1;
+	elif ! curl --version > /dev/null; then
+		echo "This script needs curl to setup your device" 1>&2
+		exit 1;
+	fi
+	if test -x /sbin/systemctl; then
```

Comment:
On my VM it's `/usr/bin/systemctl`, so this did not work. Might want to check `which systemctl` or something similar.